src/kaenguruen/load_data.py

```diff
@@ -0,0 +1,163 @@
+    Args:
+        filepath: Absolute path to the text file
+        data_dir: Base directory for calculating relative path
+        
+    Returns:
+        A dictionary with the parsed sections
+    """
+    # Initialize an empty dictionary to store problem data
+    problem_data = {}
+    
+    # Add the relative path to the dictionary
+    rel_path = os.path.relpath(filepath, data_dir)
+    problem_data["file"] = rel_path
+    
+    try:
+        # Read the file content
+        with open(filepath, 'r', encoding='utf-8') as file:
+            content = file.read()
+        
+        # Split the content by "---" separator
+        sections = content.split("---")
+        
+        # Clean and assign each section to the appropriate key
+        if len(sections) >= 1:
+            # Remove the filepath comment if present
+            question_text = sections[0]
+            if "// filepath:" in question_text:
+                question_text = question_text.split("\n", 1)[1]
+            problem_data["question"] = question_text.strip()
+        
+        if len(sections) >= 2:
+            # Extract options, removing the header
+            options_text = sections[1]
+            if "Valgmuligheder:" in options_text:
+                options_text = options_text.replace("Valgmuligheder:", "").strip()
+            problem_data["options"] = options_text.strip()
+        
+        if len(sections) >= 3:
+            # Extract answer, removing the header
+            answer_text = sections[2]
+            if "Svar:" in answer_text:
+                answer_text = answer_text.replace("Svar:", "").strip()
+            problem_data["answer"] = answer_text.strip()
+        
+        if len(sections) >= 4:
+            # Extract solution, removing the header
+            solution_text = sections[3]
+            if "Løsning:" in solution_text:
+                solution_text = solution_text.replace("Løsning:", "").strip()
+            problem_data["solution"] = solution_text.strip()
+        
+        if len(sections) >= 5:
+            # Extract percentage correct, removing the header
+            percent_text = sections[4]
+            if "Procent rigtige:" in percent_text:
+                percent_text = percent_text.replace("Procent rigtige:", "").strip()
+                # Remove the percentage sign if present and convert to float
+                percent_text = percent_text.replace("%", "").strip()
+                try:
+                    problem_data["percentage_correct"] = float(percent_text)
+                except ValueError:
+                    problem_data["percentage_correct"] = percent_text
+            else:
+                problem_data["percentage_correct"] = percent_text.strip()
+                
+    except Exception as e:
+        print(f"Error parsing file {filepath}: {e}")
+        return {"error": str(e), "file": rel_path}
+    
+    return problem_data
+
+
+def load_data_from_directory(directory_path: str) -> List[Dict[str, Any]]:
```

Comment:
would rename to `load_kaenguruen_data()` and just assume the use of the relative path.